src/main/java/duke/gui/MainWindow.java

```diff
@@ -0,0 +1,85 @@
+package duke.gui;
+
+import duke.Duke;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+/**
+ * A controller for MainWindow. Provides the layout for the other controls.
+ * <p>
+ * Adapted from <a href="https://se-education.org/guides/tutorials/javaFxPart4.html">se-education</a>
+ */
+public class MainWindow extends AnchorPane {
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+    @FXML
+    private Button helpButton;
+    private Duke duke;
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/User.png"));
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/Duke.png"));
+
+    /**
+     * Initialises the main window.
+     */
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+    }
+
+    /**
+     * Sets the duke object.
```

Comment:
From coding standard, Javadoc comments require a newline between description and param section. I've only spotted this here, in general you followed the coding standard pretty well

Reply:
Thanks for the spot!